gn2/wqflask/views.py

```diff
@@ -343,34 +323,35 @@ def commit_gn_doc():
     return render_template("gn_editor_results_page.html", **data)
 
 
-@app.route("/gnqna/hist/", methods=["GET"])
-@require_oauth2
-def get_hist_titles():
-    token = session_info()["user"]["token"].either(
-        lambda err: err, lambda tok: tok["access_token"])
-    response = monad_requests.get(urljoin(GN3_LOCAL_URL,
-                                          "/api/llm/hist/titles"),
-                                  headers={
-        "Authorization": f"Bearer {token}"
-    }
-    ).then(lambda resp: resp).either(
-        lambda x:  x.json(), lambda x: x.json())
-    return render_template("gnqa_search_history.html", **response)
-
-
-@app.route("/gnqna/hist/search/<search_term>", methods=["GET"])
+@app.route("/gnqna/hist", methods=["GET", "DELETE"])
 @require_oauth2
-def fetch_hist_records(search_term):
+def get_gnqa_history():
+    def _error_(resp):
+        return render_template("gnqa_errors.html",
+                               **{"status_code": resp.status_code,
+                                  **resp.json()})
     token = session_info()["user"]["token"].either(
```

Comment:
how do we handle the situation when a token results in an err? I expected some exception or we render an error page. Here, we'd pass the token this error to the requests_call.

Reply:
This is more of an auth question: is it possible to have a logged in user: without a token we are  already
using the @require_oauth2 decorator. Though am not sure about this.

Reply:
ah, I see. So ideally, we should never end up with an invalid token. I'm not too sure about how to handle this too. Can we add a TODO comment here?